src/Internal/Payment/PaymentContext.php

```diff
@@ -244,14 +262,71 @@ public function get_intent(): ?WC_Payments_API_Abstract_Intention {
 		return $this->get( 'intent' );
 	}
 
+	/**
+	 * Updates state transition and associated changes
+	 *
+	 * @param string $state The state.
+	 */
+	public function update_state_transition( string $state ): void {
```

Comment:
Maybe we should rename this to `log_state_transition` for now, to make it clear that the purpose of the method is to log?

Reply:
Done